odc/stats/plugins/tcw_percentiles.py

```diff
@@ -42,23 +51,22 @@ def native_transform(self, xx):
         """
         bad = enum_to_bool(xx["fmask"], ("nodata", "cloud", "shadow")) # a pixel is bad if any of the cloud, shadow, or no-data value
         bad |= xx["nbart_contiguity"] == 0 # or the nbart contiguity bit is 0
-        xx = xx.drop_vars(["fmask", "nbart_contiguity"])
         
         for band in xx.data_vars.keys():
             bad = bad | (xx[band] == -999)
 
-        tcw = sum(coeff * xx[band] for band, coeff in self.coefficients.items())
-        tcw.attrs = xx.blue.attrs
-        tcw.attrs['nodata'] = NODATA
-        
-        xx = xx.drop_vars(xx.data_vars.keys())
-        xx['wet'] = tcw.astype(np.int16)
-        xx = keep_good_only(xx, ~bad, nodata=NODATA)
-        return xx
+        yy = xx.drop_vars(self.input_bands)
+        for m in self.output_bands:
+            yy[m] = sum(coeff * xx[band] for band, coeff in self.coefficients[m].items()).astype(np.int16)
+            yy[m].attrs = xx.blue.attrs
```

Comment:
Should there be other colours besides blue be considered for other two indices? (just spitballing here - I'm not familiar with the algorithm)

Reply:
the output was supposed to inherit the `attrs` from any input band, they're all the same other than `nodata`. So I just picked blue.